src/databricks/labs/remorph/install.py

```diff
@@ -134,38 +198,158 @@ def warehouse_type(_):
                     \nDo you want to create a new one?"""
         )
         if not allow_catalog_creation:
             msg = "Catalog is needed to setup Remorph"
             raise SystemExit(msg)
 
-        logger.info(f" Creating new Catalog {catalog_name}")
+        logger.info(f" Creating new Catalog `{catalog_name}`")
         self._catalog_setup.create(catalog_name)
 
     @retried(on=[NotFound], timeout=timedelta(minutes=5))
     def setup_schema(self, catalog_name: str, schema_name: str):
         allow_schema_creation = self._prompts.confirm(
-            f"""Schema {schema_name} not found in Catalog {catalog_name}\
+            f"""Schema `{schema_name}` not found in Catalog `{catalog_name}`\
                     \nDo you want to create a new Schema?"""
         )
         if not allow_schema_creation:
             msg = "Schema is needed to setup Remorph"
             raise SystemExit(msg)
 
-        logger.info(f" Creating new Schema {catalog_name}.{schema_name}")
+        logger.info(f" Creating new Schema `{catalog_name}.{schema_name}`")
         self._catalog_setup.create_schema(schema_name, catalog_name)
 
+    def prompt_for_new_installation(self):
+        """
+        Prompts for the new Installation and returns the configuration
+        :return: MorphConfig
+        """
+        module_prompt = self._prompts.choice(
+            "Which module(s) would you like to configure:", ["transpile", "reconcile", "all"]
+        )
+
+        morph_config, reconcile_config = None, None
+        match module_prompt:
+            case "transpile":
+                morph_config = self._prompt_for_transpile_setup()
+            case "reconcile":
+                reconcile_config = self._prompt_for_reconcile_setup()
+            case "all":
+                morph_config, reconcile_config = self._prompt_for_all()
+
+        return morph_config, reconcile_config
+
+    def _prompt_for_all(self) -> tuple[MorphConfig, ReconcileConfig]:
+        return self._prompt_for_transpile_setup(), self._prompt_for_reconcile_setup()
+
+    def _prompt_for_transpile_setup(self) -> MorphConfig:
+        logger.info("\nPlease answer a few questions to configure remorph: ** transpile **")
+
+        # default params
+        catalog_name = "transpiler_test"
+        schema_name = "convertor_test"
+        ws_config = None
+
+        source = self._prompts.choice("Select the source:", list(SQLGLOT_DIALECTS.keys()))
+
+        input_sql = self._prompts.question("Enter Input SQL path (directory/file)")
+
+        output_folder = self._prompts.question("Enter Output directory", default="transpiled")
+
+        run_validation = self._prompts.confirm(
+            "Would you like to validate the Syntax, Semantics of the transpiled queries?"
+        )
+
+        if run_validation:
+            ws_config = self._configure_runtime()
+            catalog_name = self._prompts.question("Enter Catalog for Validation")
```

Comment:
can this be a choice from dictionary? or we want to create catalogs?

https://github.com/databrickslabs/blueprint?tab=readme-ov-file#single-choice-from-dictionary

Reply:
Issue created to enhance it https://github.com/databrickslabs/remorph/issues/428

Reply:
Will pick up in next release.